### PR TITLE
Adding command to add breakpoints in the debugger

### PR DESCRIPTION
**Pull request by adri09070**
Needs https://github.com/pharo-project/pharo/pull/16919 to be merged first

### CODE REVIEW

**Comment by adri09070** [comment]
@StevenCostiou

**Comment by StevenCostiou** [comment]
thanks I will look at this

File: src/NewTools-Debugger-Commands/StBreakOnStackCommand.class.st [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+StBreakOnStackCommand class >> defaultIconName [
+
+	^ #breakOnStack
+]
+
+{ #category : 'default' }
+StBreakOnStackCommand class >> defaultName [
+
+	<toolbarDebugCommand: 50>
+	^ 'Break on stack'
```
Comment (StevenCostiou) [inline src/NewTools-Debugger-Commands/StBreakOnStackCommand.class.st]: I would name it just "break"

File: src/NewTools-Debugger-Commands/StBreakOnStackCommand.class.st [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+{ #category : 'default' }
+StBreakOnStackCommand class >> defaultDescription [
+
+	^ 'Recompile method to add a breakpoint before selection and replace the compiled methods on stack by its reflective version'
+]
+
+{ #category : 'initialization' }
+StBreakOnStackCommand class >> defaultIconName [
+
+	^ #breakOnStack
```
Comment (StevenCostiou) [inline src/NewTools-Debugger-Commands/StBreakOnStackCommand.class.st]: Does this icon exists?
Why not #smallDebug for example?

**Review by StevenCostiou — CHANGES_REQUESTED** [review]

**Comment by adri09070** [comment]
done

**Comment by StevenCostiou** [comment]
The contributions are good.
We can go when tests are fixed.
There is only one related failing test, apparently there is a implemented selector that is never sent?

**Comment by adri09070** [comment]
I think this is the message send `#findContextsForMethod:`, which is implemented in the other PR that should be merged first. I saw that there is an `assert:` left that I will remove

**Comment by adri09070** [comment]
We should wait a few hours so that the build including my changes of https://github.com/pharo-project/pharo/pull/16919 has been built

**Comment by adri09070** [comment]
For me, it is good to merge

**Pull request merged**